### DISCUSSION TITLE
Graphcache: don't cache specific queries

### DISCUSSION THREAD

**Discussion by oliverdolgener** [urql-graphql/urql#2886] [Q&A]
Hey guys, we're using urql + graphcache in production for quite some time and it really works like charm.

One question we have: is it possible to exclude some queries from being written to the cache?

Reason: Some data is only used once and would otherwise clutter the cache and leads to heavy performance issues.

We thought of something like requestPolicy but for writing to the cache.

Thanks in advance!

**Comment by JoviDeCroock** [comment]
Are you seeing performance issues? If data is only used once, is it some type of mutation?

The reason behind me asking these questions is that I have yet to see a normalized cache grow that big within a session. Over repeated sessions with offline support, yes maybe, so I would love to know/see more

**Comment by oliverdolgener** [comment]
Hey Jovi, thanks for the quick response!
Indeed we see some severe performance issues with some of our users who reach cache sizes of ~150mb.
If I understand graphcache correctly all of this is being loaded into memory which causes the app to crash immediately at startup time on some weaker devices.
We really love urql, graphcache and the offline support that comes with it but this is unfortunately a very strong downside.
What we would like to blacklist from the cache is for example Product Lists as they should be fetched each time due to price changes and availability. When a user scrolls through large product lists (we use infinite scrolling, loading batches of 40 products) the cache keeps growing and growing.
I must admit that each product is a complex somehow nested object. We try to apply unique cache keys for each nested object whenever possible.

**Reply by oliverdolgener** [reply]
@JoviDeCroock Hey man, do you have any advice? We're still struggling with this problem and had to deactivate persistent caching completely until we find a solution. Thanks in advance!

**Reply by JoviDeCroock** [reply]
You could fork https://github.com/urql-graphql/urql/blob/main/exchanges/graphcache/src/default-storage/index.ts and write your own storage-adapter that is smarter about `writeData`

**Comment by Suranjan2002** [comment]
Hey, thanks in advance for reaching me I will surely see to the matter and
if I find some solutions to it I will surely update you.
